.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,28 @@
+name: static-analysis
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  static_analysis:
```

Comment:
Please do this:

1. move this job into an existing YML file instead of adding a new one
2. for an existing YML file:
    1. rename  into `ci.yml`
    2. replace `name: tests` with `name: ci`
    3. replace `name: PHP ${{ matrix.php }}` into `name: Tests ${{ matrix.php }}`

This way we'll have all CI runs in one place. Right now I have to jump through the hooks to see static analysis and tests of the same PR on the https://github.com/hamcrest/hamcrest-php/actions page.

Example result: https://github.com/minkphp/Mink/actions/runs/13197030721.